src/main/scala/uk/gov/nationalarchives/XMLCreator.scala

```diff
@@ -34,6 +34,8 @@ class XMLCreator(ingestDateTime: OffsetDateTime) {
       identifiers: List[Identifier],
       securityDescriptor: String = "open"
   ): IO[String] = IO {
+    val transferCompleteDatetime = asset.transferCompleteDatetime
+    if (transferCompleteDatetime.isAfter(ingestDateTime)) throw new Exception("'ingestDateTime' is before 'transferCompleteDatetime'!")
```

Comment:
Please don't throw exceptions, it's very non functional :smile: 
```scala
private[nationalarchives] def createOpex(
      asset: AssetDynamoTable,
      children: List[FileDynamoTable],
      assetXipSize: Long,
      identifiers: List[Identifier],
      securityDescriptor: String = "open"
  ): IO[String] = {
    val transferCompleteDatetime = asset.transferCompleteDatetime
    IO.raiseWhen(transferCompleteDatetime.isAfter(ingestDateTime))(new Exception("'ingestDateTime' is before 'transferCompleteDatetime'!")).map { _ =>
      val xml =
        <opex:OPEXMetadata xmlns:opex={opexNamespace}>
          <opex:DescriptiveMetadata>
          // ..rest of the method
```

Reply:
🤦 I completely forgot I shouldn't do that.